var/spack/repos/builtin/packages/sed/package.py

```diff
@@ -11,8 +11,16 @@
 class Sed(AutotoolsPackage, GNUMirrorPackage):
     """GNU implementation of the famous stream editor."""
     homepage = "https://www.gnu.org/software/sed/"
-    gnu_mirror_path = "sed/sed-4.2.2.tar.bz2"
+    gnu_mirror_path = "sed/sed-4.8.tar.xz"
 
+    def url_for_version(self, version):
+        if Version("4.2") <= version < Version("4.3.0"):
+            self.gnu_mirror_path = "sed/sed-{0}.tar.bz2".format(version)
+        elif version < Version("4.2"):
+            self.gnu_mirror_path = "sed/sed-{0}.tar.gz".format(version)
+        return super(Sed, self).url_for_version(version)
```

Comment:
Can you move this method below the `executables` attribute? This is just so that we'll keep metadata at the top of the class, and methods below.

Reply:
Sure. I can do that.